### PR TITLE
Sogs convo dedup + fix sogs fetching of sogs messages

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts [hunk 1]
```diff
@@ -35,9 +37,109 @@ export type OpenGroupV2InfoJoinable = OpenGroupV2Info & {
...
+  const lowerCased = server.toLowerCase();
+  let serverHost: string | undefined;
+  try {
+    const lowerCasedUrl = new window.URL(lowerCased);
+    serverHost = lowerCasedUrl.hostname; // hostname because we don't want the port to be part of this
+    if (!serverHost) {
+      throw new Error('Could not parse URL from serverURL');
+    }
+  } catch (e) {
+    // plain ip are not recognized are url, but we want to allow the,
```
Comment (yougotwill) [inline ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts:65]: allow them?

File: ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts [hunk 2]
```diff
@@ -35,9 +37,109 @@ export type OpenGroupV2InfoJoinable = OpenGroupV2Info & {
...
+  ]);
+
+  // If the server is run by Session then include all configurations in case one of the alternate configurations is used
+  if (isSessionRunOpenGroup(serverLowerCase)) {
+    serverOptions.add(defaultServerHost);
+    serverOptions.add(`http://${defaultServerHost}`);
+    serverOptions.add(`https://${defaultServerHost}`);
+    serverOptions.add(legacyDefaultServerIP);
+    serverOptions.add(`http://${legacyDefaultServerIP}`);
+    serverOptions.add(`https://${legacyDefaultServerIP}`);
```
Comment (yougotwill) [inline ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts:118]: this is beautiful 😂

File: ts/session/apis/open_group_api/sogsv3/sogsApiV3.ts [hunk 3]
```diff
@@ -455,43 +455,43 @@ export const handleBatchPollResults = async (
...
-            break;
-          case 'outbox':
-            await handleInboxOutboxMessages(subResponse.body, serverUrl, true);
-            break;
-          default:
-            window.log.error('No matching subrequest response body for type: ', responseType);
-        }
-      })
-    );
+    for (let index = 0; index < batchPollResults.body.length; index++) {
```
Comment (yougotwill) [inline ts/session/apis/open_group_api/sogsv3/sogsApiV3.ts:458]: please leave a comment explaining here

**Review by yougotwill — CHANGES_REQUESTED** [review]
Looking good! Some minor changes needed

File: ts/node/sql.ts [hunk 4]
```diff
@@ -1607,6 +1611,274 @@ function updateToLokiSchemaVersion26(currentVersion: number, db: BetterSqlite3.D
...
+
+    console.info(
+      '========> before room update:',
+      getAllV2OpenGroupRooms(db)
+        .filter(m => m.serverUrl.includes(domainNameToUse) || m.serverUrl.includes(ipToRemove))
+        .map(m => pick(m, ['conversationId', 'serverUrl', 'roomId']))
+    );
+
+    // for those with duplicates, delete the one with the IP as we want to rely on the one with the DNS only now
+    // remove the ip ones completely which are in double.
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/node/sql.ts:1705]: fixed with `duplicated`

File: ts/node/sql.ts [hunk 5]
```diff
@@ -1607,6 +1611,274 @@ function updateToLokiSchemaVersion26(currentVersion: number, db: BetterSqlite3.D
...
+    // Note: this does also remove the ones not in double, but we are recreating them just below with `saveV2OpenGroupRoom`
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/node/sql.ts:1706]: fixed

**Review by yougotwill — APPROVED** [review]
LGTM 🚀

**Pull request merged**